### PR TITLE
[DropdownMenu]: Introduce new Dropdown Menu component

### PR DESCRIPTION
**Pull request by RiinaKuu**
Jira-ticket: https://funidata.atlassian.net/browse/DS-260

- Refactor the old DropdownMenuComponent
- Expose the component to public API

Tests will be written in another PR, https://github.com/funidata/fudis/pull/405

**Known issues atm:**
- [Safari does not focus buttons when clicking by design](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#clicking_and_focus). First tried to give focus manually but that broke the whole event flow. Since there is no focus from the start, outside click does not work at all in Safari.

### CODE REVIEW

File: ngx-fudis/projects/ngx-fudis/src/lib/components/button/button.component.html [hunk 1]
```diff
@@ -1,24 +1,28 @@
 <ng-container *ngIf="tooltip">
   <button
     #buttonElement
+    fudisTooltip
+    [class]="_classList | async"
     [type]="type"
     (blur)="_handleButtonBlur($event)"
     (focus)="_handleButtonFocus($event)"
+    (click)="!disabled && buttonClick($event)"
```
Comment (MayaMarjut) [inline ngx-fudis/projects/ngx-fudis/src/lib/components/button/button.component.html:9]: !disabled here doesn't have any effect

File: ngx-fudis/projects/ngx-fudis/src/lib/components/button/button.component.html [hunk 2]
```diff
@@ -1,24 +1,28 @@
...
+    (keydown)="_handleMenuButtonKeyDown($event)"
     [attr.aria-disabled]="disabled"
-    [attr.aria-label]="_ariaLabel"
+    [attr.aria-label]="_ariaLabel | async"
     [attr.id]="_id"
-    (click)="!disabled && buttonClick($event)"
-    [ngClass]="_classList"
-    fudisTooltip
+    [attr.aria-expanded]="!asMenuButton ? null : dropdownOpen"
```
Comment (MayaMarjut) [inline ngx-fudis/projects/ngx-fudis/src/lib/components/button/button.component.html:14]: Expanded and collapsed state is not transfered properly. Now aria expanded get's state as [object object]. Seems like this state is not properly tied to handleclick function.

Comment (MayaMarjut) [reply ngx-fudis/projects/ngx-fudis/src/lib/components/button/button.component.html:14]: Currently when focusing on closed menu-button, screen readers reads it as expanded, when should be collapsed.

File: ngx-fudis/projects/ngx-fudis/src/lib/components/button/button.component.html [hunk 3]
```diff
@@ -31,20 +35,24 @@
...
     (focus)="_handleButtonFocus($event)"
+    (keydown)="_handleMenuButtonKeyDown($event)"
     [attr.aria-disabled]="disabled"
-    [attr.aria-label]="_ariaLabel"
+    [attr.aria-label]="_ariaLabel | async"
     [attr.id]="_id"
-    (click)="!disabled && buttonClick($event)"
-    [ngClass]="_classList"
+    [attr.aria-expanded]="!asMenuButton ? null : dropdownOpen"
+    [attr.aria-controls]="asMenuButton && dropdownMenuId ? dropdownMenuId : null"
```
Comment (MayaMarjut) [inline ngx-fudis/projects/ngx-fudis/src/lib/components/button/button.component.html:48]: Consider adding aria-haspopup over here

File: ngx-fudis/projects/ngx-fudis/src/lib/components/button/button.component.html [hunk 4]
```diff
@@ -31,20 +35,24 @@
 <ng-container *ngIf="!tooltip">
   <button
```
Comment (MayaMarjut) [inline ngx-fudis/projects/ngx-fudis/src/lib/components/button/button.component.html:36]: Currently tab-key doesn't work at all on Safari. Try adding tabindex=0 here, which makes button accessible with Safari browser

File: ngx-fudis/projects/ngx-fudis/src/lib/components/dropdown-menu/dropdown-menu-item/dropdown-menu-item.component.html [hunk 5]
```diff
@@ -1,54 +1,22 @@
 <button
-  *ngIf="!_parentComponent.multiselect"
   #dropdownItem
-  class="fudis-dropdown-menu-item fudis-dropdown-menu-item__focusable"
-  [attr.role]="'menuitem'"
+  class="fudis-dropdown-menu-item"
+  [class.fudis-dropdown-menu-item--disabled]="disabled || null"
+  [class.fudis-dropdown-menu-item__focusable]="!disabled"
   [attr.id]="_id"
```
Comment (MayaMarjut) [inline ngx-fudis/projects/ngx-fudis/src/lib/components/dropdown-menu/dropdown-menu-item/dropdown-menu-item.component.html:6]: As a usability improvement I would add  tabindex="-1" here, which would make menu items unfocusable by tab key, while tab key would be preserved for moving out of menu. This way user doesn't have to tab through all menu items in order to exit while only tabing forward.

File: ngx-fudis/projects/ngx-fudis/src/lib/components/button/button.component.html (hunk 1 above)
Comment (RiinaKuu) [reply ngx-fudis/projects/ngx-fudis/src/lib/components/button/button.component.html:9]: If the button is disabled we don't want click to emit any event. It is necessary attribute.

File: ngx-fudis/projects/ngx-fudis/src/lib/components/button/button.component.html (hunk 2 above)
Comment (RiinaKuu) [reply ngx-fudis/projects/ngx-fudis/src/lib/components/button/button.component.html:14]: `dropdownOpen` observable was lacking async pipe in the template, fixed.

File: ngx-fudis/projects/ngx-fudis/src/lib/components/button/button.component.html (hunk 3 above)
Comment (RiinaKuu) [reply ngx-fudis/projects/ngx-fudis/src/lib/components/button/button.component.html:49]: Good addition, added.

File: ngx-fudis/projects/ngx-fudis/src/lib/components/button/button.component.html (hunk 4 above)
Comment (RiinaKuu) [reply ngx-fudis/projects/ngx-fudis/src/lib/components/button/button.component.html:37]: Tab key is working fine for me on Safari as well.

File: ngx-fudis/projects/ngx-fudis/src/lib/components/dropdown-menu/dropdown-menu-item/dropdown-menu-item.component.html (hunk 5 above)
Comment (RiinaKuu) [reply ngx-fudis/projects/ngx-fudis/src/lib/components/dropdown-menu/dropdown-menu-item/dropdown-menu-item.component.html:6]: Didn't know this was a good practice. Good to know, added it.

**Review by MayaMarjut — DISMISSED** [review]

**Review by MayaMarjut — DISMISSED** [review]

**Review by MayaMarjut — APPROVED** [review]

**Pull request merged**